### DISCUSSION TITLE
Block variation title not displayed in block sidebar

### DISCUSSION THREAD

**Discussion by florianbrinkmann** [WordPress/gutenberg#43835] [Q&A] [answered]
### Description

I have a block with variations, and in the block inserter I see the correct title of the variation. But after adding a variation to the content, I get the block’s name instead of the variation name in the block sidebar.

It works for the social icons core block, but I cannot get it working for the variation for my custom block.

### Step-by-step reproduction instructions

I created a small example plugin with a `Variation title bug` block that has a `One variation` variation: https://github.com/florianbrinkmann/variation-title-bug
Just download the repo, add the `One variation` to a post or page and you should get the string `Variation!` in the editor but the `Variation title bug` name in the right block sidebar instead of `One Variation`.

### Environment info

WordPress 6.0.2 without Gutenberg and no other active plugins

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment, marked as answer]
I think it is due to the missing `isActive` property.
For your repository, the following code would work:

```json
{
  "variations": [
    {
      "name": "one-variation",
      "title": "One variation",
      "attributes": {
        "isVariation": true
      },
      "isActive": ["isVariation"]
    }
  ],
}
```

[The reference for variations](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-variations/) includes the following statement:

> It’s worth mentioning that setting the isActive property can be useful for cases you want to use information from the block variation, after a block’s creation. For example, this API is used in useBlockDisplayInformation hook to fetch and display proper information on places like the BlockCard or Breadcrumbs components.

**Reply by florianbrinkmann** [reply]
Thanks a lot @t-hamano, that was the problem! And, sorry for not reading the docs in detail before creating an issue 😬

**Comment by t-hamano** [comment]
At the same time, I convert from issue to **discussion** because I think this is not a Gutenberg problem.
